src/core/services-impl/src/main/java/it/geosolutions/geostore/util/SearchConverter.java

```diff
@@ -200,7 +203,7 @@ public void visit(FieldFilter filter) throws InternalErrorServiceEx {
         f.setOperator(op);
 
         if (type == Date.class) {
-            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
SimpleDateFormat is not thread-safe. Currently, you use it in a single place (which might be okay). But in multi-threaded contexts or if SearchConverter is used as a singleton, you could get concurrency issues.

Use DateTimeFormatter (Java 8+).

Reply:
Changed to use DateTimeFormatter. Thanks!